### PR TITLE
gh-81039: Add small description of f-string's "=}" to tutorial

### PR DESCRIPTION
**Pull request by slateny**
#81039

https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals

https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

### CODE REVIEW

**Review by AlexWaygood — COMMENTED** [review]
Hmm, is this the best example? The main use case for this feature is for debugging, and there doesn't really seem to be any debugging going on in this example.

File: Doc/tutorial/inputoutput.rst [hunk 1]
```diff
@@ -133,7 +133,14 @@ applies :func:`repr`::
    >>> print(f'My hovercraft is full of {animals!r}.')
    My hovercraft is full of 'eels'.
 
-For a reference on these format specifications, see
+The ``=`` specifier can be used to expand an expression and its evaluation:
```
Comment (ericvsmith) [inline Doc/tutorial/inputoutput.rst]: Shouldn't it say something about the text of the expression? "Expand an expression" doesn't sound like that, to me.

This should also mention that it's the repr that's printed.

Comment (slateny) [reply Doc/tutorial/inputoutput.rst]: I did try to condense the wording a bit for the tutorial and just rely more on the example, but on a second read through you're definitely right that it sounds off. If borrowing the wording from the what's new,

> The `=` specifier can be used to expand an expression to the text of the expression, an equal sign, then the representation of the evaluated expression.

leaving out some information gives,

> The `=` specifier can be used to expand an expression to itself and the representation of its evaluation.

Would that work, or would you prefer the original wording? I'm also not too sure what the difference is between the 'representation of an evaluation' vs just 'evaluation'. Say for `i=1`, would the representation be `i.__repr__()` and the evaluation `eval('i')`?

**Comment by slateny** [comment]
> Hmm, is this the best example? The main use case for this feature is for debugging, and there doesn't really seem to be any debugging going on in this example.

Maybe something like

```
bugs = 'roaches'
count = '13'
area = 'living room'
print(f'Debugging: {bugs=} {count=} {area=}')
...
```

And this tutorial section could possibly also use an example for printing the curly brackets too

> ... And you can escape curly brackets in f-strings by using two:
```
...
print(f'Debugging: {{{bugs=} {count=} {area=}}}')
...
```

**Comment by ambv** [comment]
The last example you provided is very nice. I would split the example of showing `{{` into a separate one so that newbies won't confuse themselves into thinking that `{=` requires triple braces.

**Comment by ericvsmith** [comment]
> The last example you provided is very nice. I would split the example of showing `{{` into a separate one so that newbies won't confuse themselves into thinking that `{=` requires triple braces.

It is a good example, except I'd make `count` an integer, not a string. And I'd omit the `{{` example entirely, because it's orthogonal to this and covered elsewhere.

**Pull request merged**